Nav mesh

BorDekker opened the pull request.

BorDekker committed:
NPC Navmesh

Created an NPC with Navmesh
Created a C# script so that the NPC will go to the player with help of the navmesh.

BorDekker committed:
Changed the NavigationNPC C# script

Made it so that the NPC will go towards a waypoint, once it reaches said waypoint it will go to another one.

It will choose a random waypoint each time from a list.

Made multiple empty game objects in the scene. Used them as waypoints for my waypoint list.

WaterWalker023 commented on Assets/Navmesh/NavigationNPC.cs:
```diff
@@ -0,0 +1,30 @@
+
+public class NavigationNPC : MonoBehaviour
+{
+    public GameObject[] waypointList;
```
```suggestion
    [SerializeField] private List<GameObject> waypointList;
```

do u know why?

WaterWalker023 commented on Assets/Navmesh/NavigationNPC.cs:
```diff
@@ -0,0 +1,30 @@
+{
+    public GameObject[] waypointList;
+    
+    public int waypointIndex = 0;
```
```suggestion
    [SerializeField] private int waypointIndex = 0;
```

WaterWalker023 commented on Assets/Navmesh/NavigationNPC.cs:
```diff
@@ -0,0 +1,30 @@
+    // Update is called once per frame
+    void Update()
+    {
+        agent.destination = Vector3.MoveTowards(transform.position, waypointList[waypointIndex].transform.position, 10);
```
geef dit een vector3 (postition) in plaats van a movetowards

BorDekker committed:
Updated code for pull reguest

Changed the GameObject array to an list.
Made both the list and int serializefield and private.
agent destination is now equal to the position of the waypoint.

WaterWalker023 commented on Assets/Navmesh/NavigationNPC.cs:
```diff
@@ -0,0 +1,31 @@
+    // Update is called once per frame
+    void Update()
+    {
+        agent.destination = waypointList[waypointIndex].transform.position;
+
+        if (agent.remainingDistance < 1)
+        {
+            waypointIndex = Random.Range(0, waypointList.Count);
```
```suggestion

        if (agent.remainingDistance < 1)
        {
            waypointIndex = Random.Range(0, waypointList.Count);
            agent.destination = waypointList[waypointIndex].transform.position;
```

BorDekker committed:
Second change for pull request

Changed the location of a code, so that it doesn't get called every frame
